### PR TITLE
[supernova] fix /b_fill handler

### PR DESCRIPTION
**Pull request by catfact**
Purpose and Motivation
----------------------

noticed that `/b_fill` commands didn't seem to be doing anything in supernova. 

checking the source revealed that the sample index into the buffer data wasn't being incremented, so i added this.

i additionally added a bounds check for the index. surprisingly this doesn't happen in other command handlers like `/b_set`. is there a non-obvious mechanism for this elsewhere, or is it possible to smash memory with OOB arguments to these commands?

Types of changes
----------------

- Bug fix (non-breaking change which fixes an issue)

Checklist
---------

- [x] All previous tests are passing
- [x] This PR is ready for review

i haven't added any unit tests, but of course would be happy to do so with a little more guidance.

### CODE REVIEW

**Comment by catfact** [comment]
anybody?

**Comment by patrickdupuis** [comment]
Thanks for this PR!

I was able to reproduce the issue, and this PR fixes things.

Sorry for the silence. We are planning to have a Dev meeting next week to discuss 3.10 and future development.

File: server/supernova/sc/sc_osc_handler.cpp [hunk 1]
```diff
@@ -2745,15 +2745,19 @@ void handle_b_fill(ReceivedMessage const & msg)
...
         osc::int32 index = it->AsInt32(); ++it;
         verify_argument(it, end);
         osc::int32 samples = it->AsInt32(); ++it;
         verify_argument(it, end);
         float value = extract_float_argument(it++);
 
-        for (int i = 0; i != samples; ++i)
+        for (int i = 0; i != samples; ++i) {
             data[index] = value;
+            if(++index >= bufSamples) { break; }
```
Comment (snappizz) [inline server/supernova/sc/sc_osc_handler.cpp]: thanks! minor code style issues here: please move `++index` into its own line, add a space between `if (`, and move `break` to a separate line

**Comment by catfact** [comment]
ok no problem, i seem to be having some qt-related build issues with 3.10 but will hopefully sort it out presently.

**Comment by catfact** [comment]
@brianlheim @snappizz how's this look to yall?

(fwiw: i see many other violations of those formatting rules just in this source file. maybe uncrustify or clang-format is called for if the rules are serious. also documenting them would be nice.)

and, er... should i fix the other lack of bounds checking? OSC is an external API after all, one can't assume every script in the world is well behaved.

for example, i can confirm that this very evil sclang script will smash memory in supernova. usually this crashes it, and on at least one occasion i managed to take down the whole jack server.

```
Server.supernova;
s = Server.local;
s.boot;
s.waitForBoot { Routine {

	x = { SinOsc.ar([218, 222]) * 0.125 }.play(s);

	b = Buffer.alloc(s, 1);
	s.sync;
	n = 100000;
	n.do({ |i| s.sendMsg("/b_set", b.bufnum, i, 1.0.rand2); });
}.play };
```

**Comment by brianlheim** [comment]
> (fwiw: i see many other violations of those formatting rules just in this source file. maybe uncrustify or clang-format is called for if the rules are serious. also documenting them would be nice.)

both points are covered in our [style guide](https://github.com/supercollider/supercollider/wiki/Code-style-guidelines) which is referenced in our main CONTRIBUTING.md.

but yes, the codebase (which has seen many contributors) doesn't have a consistent style. i am planning to add a clang-format file and reformat everything during a quiet period, hopefully sometime in the next month or so.

> and, er... should i fix the other lack of bounds checking? OSC is an external API after all, one can't assume every script in the world is well behaved.

yes, please! (in a separate PR) thank you for noticing that.

**Comment by brianlheim** [comment]
@catfact did you mean to close this? what would you like to happen with this PR?

**Comment by catfact** [comment]
@brianlheim i am closing this b/c it is based on `develop` (per instructions in CONTRIBUTING) but you have asked me to rebase on `3.10`. i can't change the base of an existing PR.

**Comment by catfact** [comment]
(ps this is a lot of faff for a tiny diff and doesn't incline me to participate further. since i opened this PR i have actually abandoned supernova for my project, which is ARM based, due to other problems with the DSP threading implementation on that platform.)

**Comment by bgola** [comment]
> @bgola please double-check your indentation.

@brianlheim did you mean to tag me here? was there something wrong with my PR?

**Comment by brianlheim** [comment]
@bgola - oops, sorry, github autocompleted the wrong username!

> ... aaand, nevermind again, somehow GH gets a little confused by that and the rebase becomes pointless. so, keeping this closed.

I have found that in that situation, changing the base gets github to recalculate the commits. If the base is already the one you want you might have to change it twice. Pretty suboptimal on GH's part.

**Pull request closed without merging**